pkg/operation/botanist/component/vpa/vpa.go

```diff
@@ -406,54 +414,60 @@ func (v *vpa) injectAPIServerConnectionSpec(deployment *appsv1.Deployment, name
 		}
 	} else {
 		deployment.Spec.Template.Spec.AutomountServiceAccountToken = pointer.Bool(false)
-		deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env,
-			corev1.EnvVar{
-				Name:  "KUBERNETES_SERVICE_HOST",
-				Value: v1beta1constants.DeploymentNameKubeAPIServer,
-			},
-			corev1.EnvVar{
-				Name:  "KUBERNETES_SERVICE_PORT",
-				Value: strconv.Itoa(kubeapiserver.Port),
-			},
-		)
-		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
-			Name: "shoot-access",
-			VolumeSource: corev1.VolumeSource{
-				Projected: &corev1.ProjectedVolumeSource{
-					DefaultMode: pointer.Int32(420),
-					Sources: []corev1.VolumeProjection{
-						{
-							Secret: &corev1.SecretProjection{
-								LocalObjectReference: corev1.LocalObjectReference{
-									Name: v.caSecretName,
+
+		// TODO(shafeeqes): Adapt admssion-controller to use kubeconfig too, after https://github.com/kubernetes/autoscaler/issues/4844 is fixed.
+		if genericTokenKubeconfigSecretName != nil {
```

Comment:
```suggestion
		if name != admissionController {
```

Let's make this more explicit here